app/(aux)/settings.js

```diff
@@ -1,60 +1,75 @@
-import {
-	FlatList,
-	Text,
-	View,
-	TouchableOpacity,
-	Animated,
-	Image,
-	Alert,
-	TextInput
-} from 'react-native';
-import { BottomSheetModal } from '@gorhom/bottom-sheet';
+import { FlatList, Text, View, TouchableOpacity, Animated, Image, Alert, TextInput } from 'react-native';
+import { BottomSheetModal, BottomSheetFlatList } from '@gorhom/bottom-sheet';
 import { styled } from 'nativewind';
+import { signOut } from 'firebase/auth';
 import { Toggle } from '../../components/Toggle';
 import { Timer } from '../../components/Timer';
 import { Button } from '../../components/Buttons';
+import { Terms } from '../../components/Terms';
+import { Post } from '../../components/Post';
+import { Camera, CameraType } from 'expo-camera';
+import * as ImagePicker from 'expo-image-picker';
 import Ionicons from '@expo/vector-icons/Ionicons';
 import { LinearGradient } from 'expo-linear-gradient';
 import { auth } from '../../backend/config';
 import { passwordValidation } from '../../backend/functions';
-import {
-	SafeAreaView,
-	useSafeAreaInsets
-} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'; 
 import { sendPasswordResetEmail } from 'firebase/auth';
 import { updatePassword } from 'firebase/auth';
 import { reauthenticateWithCredential, EmailAuthProvider } from 'firebase/auth';
-import {
-	handle,
-	backdrop,
-	SnapPoints
-} from '../../components/BottomSheetModalHelpers';
+import { handle, backdrop, SnapPoints } from '../../components/BottomSheetModalHelpers';
 import { useAuth } from '../context/auth';
+import { getHiddenPosts, writeData, readData, uploadImage } from '../../backend/firebaseFunctions';
+import { set } from 'firebase/database';
 
 const StyledView = styled(View);
 const StyledIcon = styled(Ionicons);
 const StyledText = styled(Text);
 const StyledImage = styled(Image);
+const StyledCamera = styled(Camera);
 const StyledSafeArea = styled(SafeAreaView);
 const StyledOpacity = styled(TouchableOpacity);
 const StyledInput = styled(TextInput);
 const StyledAnimatedView = styled(Animated.View);
 const StyledGradient = styled(LinearGradient);
 
 export default function Page() {
+	const [type, setType] = useState(CameraType.front);
+	const [flashMode, setFlashMode] = useState('off');
+	const cameraRef = useRef(null);
+	const [hiddenPosts, sethiddenPosts] = useState([]);
+	const [handles, setHandles] = useState('');
+	const [newFName, setNewFName] = useState('');
+	const [newLName, setNewLName] = useState('');
 	const [currentPassword, setCurrentPassword] = useState('');
 	const [newPassword, setNewPassword] = useState('');
 	const [confirmPassword, setConfirmPassword] = useState('');
+	const [newEmail, setNewEmail] = useState('');
+	const [confirmEmail, setConfirmEmail] = useState('');
 	const [selectedReminder, setSelectedReminder] = useState(
 		new Animated.Value(0)
 	);
```

Comment:
```suggestion
	const selectedReminder = useRef(new Animated.Value(0)).current;
```
this is a better way to store Animated values